### ISSUE TITLE
[BUG] New bugs in ARIA plot 

### ISSUE THREAD

**Issue by bbuzzanga** [aria-tools/ARIA-tools#411]
Two bugs encountered immediately upon running ariaPlot: 
```ariaPlot.py -f '/u/leffe-data2/buzzanga/data/VLM/Sentinel1/HR_ARIA_2024/products/S1*.nc' -b '36.75 37.225 -76.655 -75.928' -m /u/leffe-data2/buzzanga/data/Crops/HR/OSM_wmask.tif -w  /u/leffe-data2/buzzanga/data/VLM/Sentinel1/HR_ARIA_2024 -plotcoh -plotbperpcoh -makeavgoh -nt 8```

```Traceback (most recent call last):
  File "/u/leffe-data2/buzzanga/Miniconda3/envs/ARIA/bin/ariaPlot.py", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/u/leffe-data2/buzzanga/Software_InSAR/ARIA-tools_git/tools/bin/ariaPlot.py", line 273, in <module>
    main()
  File "/u/leffe-data2/buzzanga/Software_InSAR/ARIA-tools_git/tools/bin/ariaPlot.py", line 139, in main
    logging.basicConfig(level=log_level, format=ARIAtools.util.log.FORMAT)
                                                ^^^^^^^^^^^^^^^^^^
AttributeError: module 'ARIAtools.util' has no attribute 'log'
```

```Traceback (most recent call last):
  File "/u/leffe-data2/buzzanga/Miniconda3/envs/ARIA/bin/ariaPlot.py", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/u/leffe-data2/buzzanga/Software_InSAR/ARIA-tools_git/tools/bin/ariaPlot.py", line 274, in <module>
    main()
  File "/u/leffe-data2/buzzanga/Software_InSAR/ARIA-tools_git/tools/bin/ariaPlot.py", line 183, in main
    standardproduct_info.products[0], standardproduct_info.products[1], \
ValueError: too many values to unpack (expected 7)```

**Comment by bbuzzanga** [comment]
Thanks @alexfore we indeed get past these two. I found another one, you want me to put it here or open a new issue?

**Comment by alexfore** [comment]
Give me a way to replicate it (again I need the products)

**Comment by bbuzzanga** [comment]
`ariaDownload.py -o Download -b '36.75 37.225 -76.655 -75.928' -t 004 -s 20170101 -e 20170301`

First bug: 
`ariaPlot.py -f './products/S1*.nc' -b '36.75 37.225 -76.655 -75.928' -m download -w  . -plotcoh -plotbperpcoh -makeavgoh -nt 8` 

Run again with the just created mask and you'll get another bug

`ariaPlot.py -f './products/S1*.nc' -b '36.75 37.225 -76.655 -75.928' -m ./mask/OSM_wmask.msk -w  . -plotcoh -plotbperpcoh -makeavgoh -nt 8`

**Comment by alexfore** [comment]
Thanks. This is because I'm trying to remove the passing around of GDAL instances. The callee in ariaPlot which takes `args.mask` will need to be modified to do a `osgeo.gdal.Open` on the filename which is returned by `prep_mask`. I can finish this tommorow

**Comment by alexfore** [comment]
I fixed another issue with removal of a temp file outside of the block of code where it was created. Now the problem seems to be that the mask file which is written is only zeros, later raising RuntimeError when `GetStatistics` is called on a GDAL instance which only has zeros in it. Maybe @sssangha can follow up on that as he did those mask changes

**Comment by alexfore** [comment]
@sssangha can you take a look at this Exception? To recreate the observed Exception, run the following using the `fix_aria_plot` branch:
```
ariaDownload.py -o Download -b '36.75 37.225 -76.655 -75.928' -t 004 -s 20170101 -e 20170301
```
Then download water mask and make figures:
```
ariaPlot.py -f './products/S1*.nc' -b '36.75 37.225 -76.655 -75.928' -m download -w  . -plotcoh -plotbperpcoh -makeavgoh -nt 8
```
Then regenerate figures using existing mask file
```
ariaPlot.py -f './products/S1*.nc' -b '36.75 37.225 -76.655 -75.928' -m mask/esa_world_cover_2021.msk -w  . -plotcoh -plotbperpcoh -makeavgoh -nt 8
```
This raises `RuntimeError: Failed to compute statistics, no valid pixels found in sampling.`

**Comment by sssangha** [comment]
Thank's @alexfore , I pushed a new [commit](https://github.com/aria-tools/ARIA-tools/pull/413/commits/c26bda987f7c4cce70c472b1c9883f65b65b9353) with an extended description explaining my changes.

But to recap here for more transparency, I tweaked the code to in place changes existing mask files in successive calls to ariaPlot. it is recommended to specify a fresh working directory (with the `-w`) with successive calls, but a simple tweak like what I pushed can help avoid confusion and minimize unnecessary clutter with multiple output directories.

Please let me know if I should clarify anything here.

Looks like I have some pep8 issues to resolve as well.

**Comment by alexfore** [comment]
closed by ee3efecff25627bd526859490cb20bcf671dcb0c
